lib/rubocop/cop/lint/float_precision.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Identifies Float literals that lose precision when parsed.
+      #
+      # Ruby's Float class has limited precision, so some literal values
+      # cannot be represented exactly and will be rounded. This cop identifies
+      # those cases where the original literal does not match the parsed value.
+      #
+      # @example
+      #
+      #   # bad
+      #   100000000000.000000000001  # becomes 100000000000.0
+      #   10000000000000001.0        # becomes 1.0e+16
+      #   1234567890123456789e0     # becomes 1.2345678901234568e+18
+      #
+      #   # good
+      #   0.0
+      #   1.0
+      #   1.1
+      #   10000.0
+      #   1_000.0
+      class FloatPrecision < Base
+        MSG = 'Float loses precision.'
```

Comment:
I think it would be helpful to include what the value actually gets downcasted to in the emssage in case you don't really care and just want to fix the offense.